tests/Regression/GitHub/2591/SeparateClassPreserveTest.php

```diff
@@ -14,13 +14,15 @@ public function testOriginalGlobalString()
 
     public function testChangedGlobalString()
     {
-        $GLOBALS['globalString'] = 'Hello!';
-        $this->assertEquals('Hello!', $GLOBALS['globalString']);
+        $value = 'Hello! I am changed from inside!';
+
+        $GLOBALS['globalString'] = $value;
+        $this->assertEquals($value, $GLOBALS['globalString']);
     }
 
     public function testGlobalString()
     {
-        $this->assertEquals('Hello!', $GLOBALS['globalString']);
+        $this->assertEquals('Hello', $GLOBALS['globalString']);
```

Comment:
$GLOBALS can't be shared between the methods, due to different processes. This was passing before because @runClassInSeparateProcess didn't have any effects, and each test was executed in master thread.